geopandas/geodataframe.py

```diff
@@ -506,6 +512,16 @@ def crs(self, value):
                 "geometry column.",
             )
 
+        if self.crs is not None:
+            warnings.warn(
+                f"Overriding the CRS of a GeoDataFrame that "
+                f"already has CRS: {self.crs}. "
+                f"This unsafe behavior will be deprecated in future versions."
+                f"Use GeoDataFrame.set_crs method instead!",
+                stacklevel=1,
+                category=DeprecationWarning,
```

Comment:
```suggestion
            warnings.warn(
                f"Overriding the CRS of a GeoDataFrame that "
                f"already has CRS. "
                f"This unsafe behavior will be deprecated in future versions."
                f"Use GeoDataFrame.set_crs method instead.",
                stacklevel=2,
                category=DeprecationWarning,
```

1. CRS printing can be veery noisy
2. no need to shout :)
3. proper stacklevel

please replicate the same in GeoSeries.